nautobot/extras/forms/mixins.py

```diff
@@ -137,6 +140,49 @@ def __init__(self, *args, **kwargs):
             self.custom_fields.append(field_name)
 
 
+class NoteModelBulkEditFormMixin(BulkEditForm):
+    """Bulk-edit form mixin for models that support Notes."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.obj_type = ContentType.objects.get_for_model(self.model)
+
+        self.fields["object_note"] = CommentField()
+        self.fields["object_note"].label = "Note"
+
+    def save_note(self, *, instance, user):
+        if "object_note" in self.cleaned_data:
+            value = self.cleaned_data.get("object_note")
+            note = Note.objects.create(
+                note=value,
+                assigned_object_type=self.obj_type,
+                assigned_object_id=instance.pk,
+                user=user,
+            )
+            logger.debug("Created %s", note)
+
+
+class NoteModelFormMixin(forms.ModelForm):
+    def __init__(self, *args, **kwargs):
+        self.obj_type = ContentType.objects.get_for_model(self._meta.model)
+
+        super().__init__(*args, **kwargs)
+
+        self.fields["object_note"] = CommentField()
+        self.fields["object_note"].label = "Note"
+
+    def save_note(self, *, instance, user):
```

Comment:
Any chance we can consolidate these two form classes? They look like they're nearly identical.

Reply:
Same with RelationshipModel... The differences are the trick here.  They have different base forms (tested) and different methods for accessing the model.

Reply:
I think I got it though, running tests.